### DISCUSSION TITLE
ArtQRCode finderPatternImage doesn't work !

### DISCUSSION THREAD

**Discussion by Flo230277** [Shane32/QRCoder#575] [Q&A]
I tried to use but it doesn't work.

Please Help.

`var bmp = new ArtQRCode(data).GetGraphic(10, Color.Black, Color.White, Color.Transparent, drawQuietZones: true, finderPatternImage: (Bitmap)Image.FromFile(logo), backgroundImage : (Bitmap)Image.FromFile(logo));`

![test](https://github.com/user-attachments/assets/81e92c26-e206-4170-b7ce-6217cdcd3827)

**Comment by codebude** [comment]
Can you give the full code for QR creation? Which ECC level do you use? Habe you tried ECCLevel.H? (Highest recovery.) Using ART QR codes is a bit tricky as every "artsy" modification lowers the readability of a code. The more you modify the QR the hard it gets to read. Rasing the ECC (error correction level) may help up to a certain level. If it's still not readable with ECCLevel H, try to use logos and patterns less abstract.

**Reply by Flo230277** [reply]
Hello,

Here is my entire code :

`var gen = new QRCodeGenerator();
            var logo = HttpContext.Current.Server.MapPath("~/") + "play.png";
            var data = gen.CreateQrCode("This is a quick test! 123#?", QRCodeGenerator.ECCLevel.H);
            var bmp = new ArtQRCode(data).GetGraphic(10, Color.Black, Color.White, Color.Transparent, drawQuietZones: true, finderPatternImage: (Bitmap)Image.FromFile(logo));
            bmp.Save(HttpContext.Current.Server.MapPath("~/") + "test.png", ImageFormat.Png);`

![test](https://github.com/user-attachments/assets/51f9c992-2d0b-46ab-9b63-7090c1a62ebc)


Logo test : 


![play](https://github.com/user-attachments/assets/f90e7deb-e264-4c96-919d-9feaeaea02ef)

**Comment by Flo230277** [comment]
Do you have a solution ?
